src/Promitor.Agents.Scraper/config/metrics-declaration.yaml

```diff
@@ -0,0 +1,33 @@
+version: v1
+azureMetadata:
+  tenantId: 5766f879-f51d-4e43-88e6-a8c609103041
+  subscriptionId: 4d47ec28-1799-4c7b-9e34-81300079de6b
+  resourceGroupName: ag1-data-platform-core
+  cloud: "UsGov"
+metricDefaults:
+  aggregation:
+    interval: 00:00:01
+  limit: 10
+  labels:
+    geo: china
+    environment: dev
+  scraping:
+    # Every minute
+    schedule: "0 * * ? * *"
+metrics:
+  - name: azure_storage_account_capacity_discovery
+    description: "The average capacity used in the storage account"
+    resourceType: StorageAccount
+    azureMetricConfiguration:
+      metricName: UsedCapacity
+      aggregation:
+        type: Average
+    resources: # Optional, required when no resource discovery is configured
+    - accountName: platformdatalaketest
+  - name: azure_logs_analytics
+    description: "Query 1 cai gi do"
+    resourceType: LogAnalytics
+    logAnalyticsConfiguration:
+      query: AzureActivity | take 100
```

Comment:
I'm not sure if this is representable as this should output a metric and not a table? Can you come up with a reasonable example please? This would help me understand if this configuration proposal is aligned with the needs

Reply:
For example:
- Should this be part of `metrics` or do we introduce a new `logs` section instead?
- Should `logAnalyticsConfiguration` be in the root of this object or part of the resource configuration? If so, do we require duplication or introduce re-usable scripts?

Reply:
In my opinion, **query** is enough if we have conditions for it
- **query** can just query 1 column and must to have alias `result` for that column
- **query** result is just 1 row and it's value need to be float
Example:
      query: Usage | take 1 | extend result = Quantity | project result

Reply:
>Should this be part of metrics or do we introduce a new logs section instead?
I want it to be part of `metrics` because in the end, I still want to expose it to some metric sink like PrometheusSink, StatsdSink

Reply:
>Should logAnalyticsConfiguration be in the root of this object or part of the resource configuration? If so, do we require duplication or introduce re-usable scripts?

I think it's better to be in the root of this object. It will be the same thing as azureMetricConfiguration. If resourceType = 'LogAnalytics' then config object must contain `logAnalyticsConfiguration` field. Else it must contain ` azureMetricConfiguration`

But yeah we can also put this `query` field in `resources` but I just think `query` is not really belong to `resources`

Reply:
I get the reasoning for that but that's also why I'm wondering if adding it under metrics is the best place but let's keep it here for now.

Can you add validation so that there is only one output column which is called result? Do you know if there is a .NET lib to parse Kusto?

Reply:
We can use this lib https://github.com/microsoft/Kusto-Query-Language/blob/master/src/Kusto.Language/readme.md
I think for the  logAnalyticsConfiguration config, we should have 2 more field: table and schema. So it will help us to use this lib to validate the query better

Reply:
Something like this 
```
   logAnalyticsConfiguration:
        table: Usage
        schema: "(Quantity: real)"
        query: Usage | take 1 | extend result = Quantity | project result
        logAnalyticsAggregation:
               interval: 100:00:00:00
```

Reply:
What do you think about this option ? Because those new data will just help us for more strict validation. The data we need for querying logAnalyticsAggregation is still just `query` field

Reply:
Why do we need the tables and schema field? I'm not sure if I get that. Is that just because of the  Kusto.Language lib?

I think it's overkill to be honest and we should just have query